### PR TITLE
Terraform AWS

### LINKED ISSUE #78: CI/CD for AWS deployments
Related to #69 and #77

# Description

Write a CI job that demonstrates the AWS deployment target for CCO.

- [need credentials for a real sandbox account on AWS for these tests]
- use terraform default configuration to set up environment required by CCO on AWS:
   - Create RDS instance
   - Create EKS cluster
   - Create IAM roles if necessary
- ckan-cloud-operator cluster initialize should run cleanly
   - no extra step or configuration should be required out of terraform + cco cluster init
- deploy a ckan instance with all required services
- write tests that show CKAN is responding as expected for read and and writes
   - Same ones as the ones running in #77 
- teardown the environment on AWS

# Acceptance criteria

- [x] Deployment of a CKAN instance into the cluster is green
- [x] CCO and its AWS dependencies are deployed into the GSA sandbox via CircleCI

### LINKED ISSUE #101: Restore unit tests

### PR DESCRIPTION
**Pull request by akariv**
Provision an EKS cluster and all peripheral services on AWS.

Fixes #78 
Fixes #101

### CODE REVIEW

**Comment by akariv** [comment]
@zelima 
@aluminiumgeek 
ping

**Comment by OriHoch** [comment]
:+1:

File: .travis.sh [hunk 1]
```diff
@@ -17,6 +17,24 @@ elif [ "${1}" == "install-tools" ]; then
...
+      echo Installing Terraform
+      (cd terraform/aws/ && \
+        curl -o tf.zip https://releases.hashicorp.com/terraform/0.12.18/terraform_0.12.18_linux_amd64.zip && \
+        unzip tf.zip && \
+        ./terraform version)
+
+      # Install AWS tools
+      echo Installing AWS tools
+      pip install awscli
+      curl -o aws-iam-authenticator https://amazon-eks.s3-us-west-2.amazonaws.com/1.14.6/2019-08-22/bin/linux/amd64/aws-iam-authenticator
```
Comment (aluminiumgeek) [inline .travis.sh]: Version of `aws-iam-authenticator` could be defined as a constant on top of the file.

File: .travis.sh [hunk 2]
```diff
@@ -17,6 +17,24 @@ elif [ "${1}" == "install-tools" ]; then
       echo Minikube Installed Successfully!
     fi
 
+    if [ "${K8_PROVIDER}" == "aws" ]; then
+      # Install terraform
+      echo Installing Terraform
+      (cd terraform/aws/ && \
+        curl -o tf.zip https://releases.hashicorp.com/terraform/0.12.18/terraform_0.12.18_linux_amd64.zip && \
```
Comment (aluminiumgeek) [inline .travis.sh]: Terraform version could be defined as a constant on top of the file.

File: ckan_cloud_operator/logs.py [hunk 3]
```diff
@@ -74,9 +74,10 @@ def log_subprocess_output(stdout, stderr):
 def subprocess_run(command, input=None):
     completed = subprocess.run(
         command, input=input, 
-        shell=True, check=True, capture_output=True
+        shell=True, check=False, capture_output=True
     )
     log_subprocess_output(completed.stdout, completed.stderr)
+    completed.check_returncode()
```
Comment (aluminiumgeek) [inline ckan_cloud_operator/logs.py:80]: Good catch!

File: .travis.sh (hunk 2 above)
Comment (akariv) [reply .travis.sh]: Done

File: .travis.sh (hunk 1 above)
Comment (akariv) [reply .travis.sh]: Done

File: ckan_cloud_operator/logs.py (hunk 3 above)
Comment (akariv) [reply ckan_cloud_operator/logs.py:80]: Thanks

**Review by aluminiumgeek — APPROVED** [review]

**Pull request merged**